model/ArrayList.php

```diff
@@ -360,6 +360,7 @@ protected function parseSortColumn($column, $direction = null) {
 		}
 
 		// Parse column specification, considering possible ansi sql quoting
+		$column = str_replace('"', '', $column); // Remove quotes so sorts like '"SiteTree"."Sort"' work.
 		if(preg_match('/^"?(?<column>[^"\s]+)"?(\s+(?<direction>((asc)|(desc))(ending)?))?$/i', $column, $match)) {
```

Comment:
Perhaps you need another capture group to remove (and ignore) the table portion of the column sort? That could work.